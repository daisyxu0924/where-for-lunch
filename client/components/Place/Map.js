import React from 'react';
import PropTypes from 'prop-types';
import styles from './Place.css';

class Map extends React.PureComponent {
  componenDidMount() {
    this.initMap();
  }

  initMap() {
    console.log('google', google);
    google.maps.Map(document.getElementById('map'), { zoom: 8, center: this.props.coordinates });
  }

  render() {
    return (
      <div id="map" className={styles.map}></div>
    );
  }
}

Map.propTypes = {
  coordinates: PropTypes.object.isRequired,
};

export default Map;
