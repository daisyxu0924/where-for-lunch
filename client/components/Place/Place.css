@import 'styles/variables.css';

.name {
  margin-top: calc($base-spacing * 10);
  min-height: 2rem;
  font-size: 2rem;
  margin-bottom: calc($base-spacing * 2);
}
.box {
  min-height: 10rem;
  background-color: white;
  color: $black;
  background: linear-gradient(to bottom, rgba(255,255,255,1) 0%,rgba(255,255,255,0) 100%);
  border-radius: 1rem;
  padding: calc($base-spacing * 2);
  > div {
    margin-bottom: calc($base-spacing / 2);
  }
  .rating {
    margin-top: $base-spacing;
    position: realtive;
    display: inline;
  }
  .ratingScore {
    display: inline-block;
  }
}
