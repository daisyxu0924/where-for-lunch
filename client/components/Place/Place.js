import React from 'react';
import PropTypes from 'prop-types';
import Stars from 'components/Stars/Stars';
import styles from './Place.css';

const Place = ({ place }) => {
  return (
    <div className={styles.root}>
      <div className={styles.name}>{ place.name || 'Where for lunch?' }</div>
      <div className={styles.box}>
        <div>{ place.address }</div>
        <div>{ place.phone }</div>
        <div>{ place.categories && place.categories.join(', ') }</div>
        <div>{ place.price }</div>
        {place.rating && <div className={styles.ratingScore}>{ place.reviewCount } reviews</div>}
        <Stars rating={place.rating}/>
      </div>
    </div>
  );
};

Place.propTypes = {
  place: PropTypes.object,
};

export default Place;
