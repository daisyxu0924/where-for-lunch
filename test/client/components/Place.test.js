import React from 'react';
import { shallow } from 'enzyme';
import Place from 'client/components/Place/Place';
import renderer from 'react-test-renderer';

describe('Place Component', () => {
  test('With Enzyme, doest not render rating section when no rating passed over', () => {
    const place = { hehe: 'haha' };
    const wrapper = shallow(
      <Place place={place} />,
    );
    const p = wrapper.find('.rating');
    expect(p.length).toBe(0);
  });

  test('With Jest snapshot, renders rating section when present', () => {
    const place = { hehe: 'haha', rating: 3.5 };
    const placeComponent = renderer
      .create(<Place place={place} />)
      .toJSON();
    expect(placeComponent).toMatchSnapshot();
  });
});
