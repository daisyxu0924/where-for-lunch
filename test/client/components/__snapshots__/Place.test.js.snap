// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Place Component With Jest snapshot, renders rating section when present 1`] = `
<div
  className="root"
>
  <div
    className="name"
  >
    Where for lunch?
  </div>
  <div
    className="box"
  >
    <div />
    <div />
    <div />
    <div />
    <div
      className="ratingScore"
    >
       reviews
    </div>
    <span
      className="root"
    >
      <div
        className="rating"
      >
        <div
          className="stars"
        >
          <div
            className="emptyStars"
          />
          <div
            className="fullStars"
            style={
              Object {
                "width": "70%",
              }
            }
          />
        </div>
      </div>
    </span>
  </div>
</div>
`;
